eos_config: fails on earlier EOS versions due to session-config

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
eos_config

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
I believe this is unchanged in 2.3 from my quick read of the source, I have not however tested it.
##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
EOS 4.14.12M 

##### SUMMARY
EOS config attempts to use a configuration session in a way that isnt supported on older EOS versions. Specifically the "show session-config diffs" command doesnt always exist.

##### STEPS TO REPRODUCE
On older versions of EOS specifically tested against 4.14.12M this play will fail
```yaml
---
- name: Change the password
  eos_config:
    provider: "{{ cli }}"
    force: true
    lines:
      - "username admin privilege 15 role network-admin secret 0 {{ secret_password }}"
```
##### EXPECTED RESULTS
Successful run of the play

##### ACTUAL RESULTS
```
fatal: [<removed>]: FAILED! => {
    "changed": false,
    "commands": [
        "configure session ansible_1482233620",
        "show session-config diffs",
        "end"
    ],
    "failed": true,
    "invocation": {
        "module_args": {
            "after": null,
            "auth_pass": null,
            "authorize": false,
            "backup": false,
            "before": null,
            "config": null,
            "defaults": false,
            "force": true,
            "host": <removed>,
            "lines": [
                "username admin privilege 15 role network-admin secret 0 <removed>",
            ],
            "match": "none",
            "parents": null,
            "password": <removed>
            "port": null,
            "provider": {
                "host": <removed>
                "password": <removed>
                "transport": "cli",
                "username": <removed>
            },
            "replace": "line",
            "save": false,
            "src": null,
            "ssh_keyfile": null,
            "timeout": 10,
            "transport": "cli",
            "use_ssl": true,
            "username": <removed>,
            "validate_certs": true
        }
    },
    "msg": "matched error in response: show session-config diffs\r\n% Invalid input\r\n<removed>(config-s-ansibl)#"
}
```

I am facing similar issue with ansible 2.5


- hosts:
    - "{{ switches }}"
  serial: 1
  gather_facts: no
  connection: local


  vars_prompt:
    - name: "password"
      prompt: "Enter user password"
      private: yes
      no_log: True

  vars:
    provider:
      host: "{{ inventory_hostname }}"
      username: "{{ ansible_user }}"
      password: "{{ password }}"

  tasks:

    - name: "Running pre-change confirmations"
      eos_command:
       provider: "{{ provider }}"
       commands: "show vlan | grep -w  {{ id }}"
       authorize: yes
      register: output

    - name: "Print pre-change output"
      debug: var=output.stdout_lines

#    - name: "Verifying the VLAN does not already exist"
#      fail: msg="[ERROR] VLAN {{ id }} already exists on {{ inventory_hostname }} Please verify the correct VLAN ID is being used. Aborting..."
#      when: output.stdout[0] != ""

    - name: " Pause for Implementor acknowledgement "
      pause:
        prompt: "Press 'Enter' to add the vlan"

    - name: "Adding the requested vlans"
      eos_config:
       provider: "{{ provider }}"
       authorize: yes
       lines:
          - vlan {{ id }}

    - name: "Running post-change confirmations"
      eos_command:
       provider: "{{ provider }}"
       commands: "show vlan | grep -w {{ id }}"
       authorize: yes
      register: output

    - name: "Print post-change output"
      debug: var=output.stdout_lines

#    - include: ../home/fds/netteam/arista-ansible/save_config.yml

    - name: "Pause for 50 seconds before proceeding to the next device.\r\nIf there is an STP event, abort the script!"
      pause:
        seconds: 50
ACTUAL RESULT

12:38:41[mshaik02@sysengb03:/home/fds/netteam/arista-ansible] ansible-playbook vlanadd.yml --extra-vars "id=125, switches=ct-bigip-lab"
Enter user password: 
 [WARNING]: Could not match supplied host pattern, ignoring: ct-bigip-lab

PLAY [ct-bigip-lab] *******************************************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY RECAP ****************************************************************************************************************************************************************************************************************

12:38:59[mshaik02@sysengb03:/home/fds/netteam/arista-ansible] ansible-playbook vlanadd.yml --extra-vars "id=125, switches=CT-7500-LAB"
Enter user password: 

PLAY [CT-7500-LAB] ********************************************************************************************************************************************************************************************************

TASK [Running pre-change confirmations] ***********************************************************************************************************************************************************************************
ok: [CT-7500-LAB]

TASK [Print pre-change output] ********************************************************************************************************************************************************************************************
ok: [CT-7500-LAB] => {
    "output.stdout_lines": [
        [
            "125   VLAN0125                         active"
        ], 
        [
            ""
        ]
    ]
}

TASK [Pause for Implementor acknowledgement] ******************************************************************************************************************************************************************************
[Pause for Implementor acknowledgement]
Press 'Enter' to add the vlan:

ok: [CT-7500-LAB]

TASK [Adding the requested vlans] *****************************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: CT-7500-LAB(s2)#
fatal: [CT-7500-LAB]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_H3uFBl/ansible_module_eos_config.py\", line 491, in <module>\n    main()\n  File \"/tmp/ansible_H3uFBl/ansible_module_eos_config.py\", line 414, in main\n    response = load_config(module, commands, replace=replace, commit=commit)\n  File \"/tmp/ansible_H3uFBl/ansible_modlib.zip/ansible/module_utils/network/eos/eos.py\", line 472, in load_config\n  File \"/tmp/ansible_H3uFBl/ansible_modlib.zip/ansible/module_utils/network/eos/eos.py\", line 241, in load_config\n  File \"/tmp/ansible_H3uFBl/ansible_modlib.zip/ansible/module_utils/connection.py\", line 146, in __rpc__\nansible.module_utils.connection.ConnectionError: show session-config diffs\r\n% Invalid input\r\nCT-7500-LAB(s2)#\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}

PLAY RECAP ****************************************************************************************************************************************************************************************************************
CT-7500-LAB                : ok=3    changed=0    unreachable=0    failed=1

--- a/changelogs/fragments/56407_eos_use_sessions.yaml
+++ b/changelogs/fragments/56407_eos_use_sessions.yaml
@@ -1,0 +1,2 @@
+minor_changes:
+  - eos_use_sessions is now type boolean instead of int.

--- a/lib/ansible/plugins/cliconf/eos.py
+++ b/lib/ansible/plugins/cliconf/eos.py
@@ -226,8 +226,10 @@ class Cliconf(CliconfBase):
             if self._session_support:
                 return self._session_support
 
-            response = self.get('show configuration sessions')
-            self._session_support = 'error' not in response
+            try:
+                self.get('show configuration sessions')
+            except AnsibleConnectionFailure:
+                self._session_support = False
 
         return self._session_support
 

--- a/lib/ansible/plugins/cliconf/eos.py
+++ b/lib/ansible/plugins/cliconf/eos.py
@@ -30,8 +30,8 @@ description:
 version_added: "2.4"
 options:
   eos_use_sessions:
-    type: int
-    default: 1
+    type: boolean
+    default: yes
     description:
       - Specifies if sessions should be used on remote host or not
     env:
@@ -214,13 +214,7 @@ class Cliconf(CliconfBase):
         return diff
 
     def supports_sessions(self):
-        use_session = self.get_option('eos_use_sessions')
-        try:
-            use_session = int(use_session)
-        except ValueError:
-            pass
-
-        if not bool(use_session):
+        if not self.get_option('eos_use_sessions'):
             self._session_support = False
         else:
             if self._session_support:

